### ISSUE TITLE
need a plugin that adds a autocomplete feature for c++ 

### ISSUE THREAD

**Issue by PrabhatProxy** [rxi/lite#231]
I want a c++  autocomplete feature like in vs-code or code::blocks

**Comment by moje01konto** [comment]
autor forget the project

**Comment by Tmpod** [comment]
He didn't forget it, he's just busy with other stuff.

Either way, I highly doubt rxi would want to add autocompletion to lite, as it is far from a trivial thing. You should close this issue and open a request at the plugins repo.

**Comment by drmargarido** [comment]
rxi added support for that like in the middle of last year. It's simple to add suggestions to the autocomplete.
If people want they can add plugins to do it. Here is a demo he did by using ctags -> https://cdn.discordapp.com/attachments/404399251276169217/709474818042364034/complete.gif

**Comment by Tmpod** [comment]
I meant language-specific autocompletions, but yeah, plugins is the way to go here.

**Comment by Tmpod** [comment]
By the way, are you going to release that ctags plugin? I'd be really interested in using it and potentially contribute to it too.

**Comment by drmargarido** [comment]
This was a gif from rxi displayed last year somewhere around here. I can build something similar when I have some free time.

**Comment by einthusan** [comment]
Golang uses a language server called gopls, i think VSCode is also using this, and there is a move towards a standard and more efficient mechanism for auto completion where the editor only provides a connection and gui but does not need to actually implement the language specific details of auto completion. Would be nice to move off Vim and try this editor. How does this compare to vim, just curious.

**Comment by Tmpod** [comment]
Yeah that's the LSP (Language Server Protocol), something VSCode popularized.

I believe there's a plugin somewhere that was implementing an LSP client for lite, but I don't seem to have it stored on my history and I can't find it online :/

lite is super extensible though and it's rather easy to make plugins. You could make an autocomplete plugin that would utilize an already existing LSP client written in Lua, I suppose.

Either way, you should checkout [lite-xl]([https://lite-xl.github.io](https://lite-xl.github.io/en/plugins/repository/)/), an active fork of lite.
